### ISSUE TITLE
[feature] Pass cli args via a configuration file

### ISSUE THREAD

**Issue by Fi5t** [skylot/jadx#1731]
*Describe your idea:*

Hello! It would be nice to pass cli arguments to jadx via a config file instead of flags. jadx-gui already has the ability to copy existing settings to the clipboard. I think this function can be used to create config files for jadx-cli. What do you think?

**Comment by skylot** [comment]
@Fi5t thanks for bringing up this topic. There are several implementations and issues:
Now jadx-cli already support ["@ syntax" from jcommander library](https://jcommander.org/#_syntax), so it is possible to store cli flags in file.  
Example with usage notes:
```bash
-v
# split option and value
-d
out-jadx-tmp
# remove quotes for string values
--rename-flags
none

classes.dex
```
and use like:
```bash
jadx @config
```
It is possible to implement generation of this file in jadx-gui to export settings. But taking into account that there are other options stored in jadx project file like renames (#1341) and comments, maybe it will better just support jadx project file in jadx-cli. Main issue in that approach that jadx-gui uses 2 configs: project and global settings, so looks like export for jadx-cli will needed anyway.
So in short: we need to implement import of jadx project file with embedded global settings into jadx-cli and add export of such file in jadx-gui.

P.S this will also help to resolve #534 (support excluded packages in jadx-cli)

**Comment by skylot** [comment]
@Fi5t I commit implementation for config load/save in jadx-cli :tada:
New options:
```
--config <config-ref>      - load configuration from file, <config-ref> can be:
                              path to '.json' file
                              short name - uses file with this name from config directory
                              'none' - to disable config loading
--save-config <config-ref> - save current options into configuration file and exit, <config-ref> can be:
                              empty - for default config
                              path to '.json' file
                              short name - file will be saved in config directory
```
Examples:
- save options into default config: 
  `jadx --threads-count 2 --save-config`
- load config from jadx-gui and save as default config for jadx-cli:
  `jadx --config gui --save-config`
-----
Also, added option to get current config directory (and other directories used by jadx):
```
--print-files     - print files and directories used by jadx (config, cache, temp)
```
on Linux it will output:
```
Files and directories used by jadx:
 - default config file: /home/<user>/.config/jadx/cli.json
 - config directory:    /home/<user>/.config/jadx
 - cache directory:     /home/<user>/.cache/jadx
 - temp directory:      /tmp
```
-----
You can check these changes using [latest unstable build](https://nightly.link/skylot/jadx/workflows/build-artifacts/master).
Feel free to report any issues or suggestions :slightly_smiling_face:

**Comment by Fi5t** [comment]
@skylot thank you! Really nice feature. I've successfully migrated my previous config using this command:
```shell
jadx @good-old-config.jadx --save-config
```
